client/src/main/java/org/apache/cassandra/sidecar/client/SidecarClient.java

```diff
@@ -68,6 +69,26 @@ public SidecarClient(SidecarInstancesProvider instancesProvider,
         executor = requestExecutor;
     }
 
+    /**
+     * Executes the Sidecar health request using the default retry policy and configured selection policy
+     *
+     * @return a completable future of the Sidecar health response
+     */
+    public CompletableFuture<HealthResponse> sidecarHealth()
+    {
+        return executor.executeRequestAsync(requestBuilder().sidecarHealthRequest().build());
```

Comment:
I think for these calls we need a custom retry policy. We probably want to limit the number or retries, I think we also need to support sidecar health checks on a specific instance, and a list of instances

Reply:
Rather than assume a retry policy and hardcoded in the client, what do you think about this?

```java
    public CompletableFuture<HealthResponse> sidecarHealth(Consumer<RequestContext.Builder> requestOverrides)
    {
        RequestContext.Builder builder = requestBuilder().sidecarHealthRequest();
        requestOverrides.accept(builder);
        return executor.executeRequestAsync(builder.build());
    }
```

Reply:
As we discussed, changed this to perform no retries.

Reply:
Added a new `OncePerInstanceRetryPolicy` as per Francisco's suggestion.